src/senaite/core/browser/fields/record.py

```diff
@@ -246,7 +247,7 @@ def set(self, instance, value, **kwargs):
         ObjectField.set(self, instance, value, **kwargs)
 
     def _to_dict(self, value):
-        if type(value) != type({}) and hasattr(value, 'keys'):
+        if not isinstance(value, dict) and hasattr(value, 'keys'):
             new_value = {}
             new_value.update(value)
             return new_value
```

Comment:
I'd like to leave a comment on the next line (253/254) but GitHub won't let me. 🙄

Do we really want to return `value` here if it is not an instance of `dict`?

Reply:
Good question and no idea at the moment. Anyways, these Archetype fields will be all removed as soon as we have migrated all contents/fields to Dexterity.